slackclient/slackrequest.py

```diff
@@ -74,18 +75,26 @@ def do(self, token, request="?", post_data=None, domain="slack.com", timeout=Non
         # use the 'file' argument to point to a File ID.
         post_data = post_data or {}
 
-        # Check for plural fields and convert them to comma-separated strings if needed
-        for field in {'channels', 'users', 'types'} & set(post_data.keys()):
-            if isinstance(post_data[field], list):
-                post_data[field] = ",".join(post_data[field])
-
         # Move singular file objects into `files`
         upload_requests = ['files.upload']
 
+        # Move file content into requests' `files` param
         files = None
         if request in upload_requests:
             files = {'file': post_data.pop('file')} if 'file' in post_data else None
 
+        # Check for plural fields and convert them to comma-separated strings if needed
+        for field in {'channels', 'users', 'types'} & set(post_data.keys()):
+            if isinstance(post_data[field], list):
+                post_data[field] = ",".join(post_data[field])
+
+        # Convert any non-String params to JSON
```

Comment:
what about integer types? or boolean types? we don't want them converted to JSON, right?

Reply:
This restores functionality that was accidentally removed: 
https://github.com/slackapi/python-slackclient/commit/7a11a49a2307b622f301352c12e1182c9cc07907#diff-69c59ad8e2680e7aaf31145d10eecab2L72

Reply:
so i guess JSON encoding those other types will "just work"? can we throw one test in there with some non-string types and verify the output?

Reply:
Made it a little more explicit